routetable/route_table.go

```diff
@@ -552,11 +613,16 @@ func (r *RouteTable) syncRoutesForLink(ifaceName string) error {
 				Scope:     netlink.SCOPE_LINK,
 			}
 
+			if target.Proto != RTProtocolUnspec {
+				logCxt.Infof("setting route protocol to %d", target.Proto)
+				route.Protocol = int(target.Proto)
+			}
+
 			if target.GW != nil {
 				route.Gw = target.GW.AsNetIP()
 			}
 
-			if target.Type == TargetTypeVXLAN {
+			if target.Type == TargetTypeVXLAN || target.Type == TargetTypeDefault {
```

Comment:
Hm, I'm not sure we need to set these for non-VXLAN routes. Have you tried it?

Reply:
I did and it wasn't working, but it may have been something else causing the problem. I'll try again.

Reply:
I had to keep this in because it would break with it, the routes couldn't be added